facebook/docusaurus
# feat(v2): support custom description for blog-only mode

## Motivation

In blog-only mode, the meta description is set to "Blog", which leads to poor SEO. The custom homepage under `src/pages/index.js` uses the value of a `customFields.description`field defined in `docusaurus.config.js` to display the description. In this PR, I use the same `customFields.description` config value for the `BlogPostList` component in the docusaurus-theme-classic theme.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

1. In `docusaurus.config.js`, look at the `customFields.description` value：

```javascript
customFields: {
    description:
      'An optimized site generator in React. Docusaurus helps you to move fast and write content. Build documentation websites, blogs, marketing pages, and more.',
  }
```
2. Set the site to blog-only mode by setting `routeBasePath` to `'/'` in `presets`
```javascript
presets: [
    [
      '@docusaurus/preset-classic',
      {
        blog: {
          path: '../website-1.x/blog',
          postsPerPage: 3,
          feedOptions: {
            type: 'all',
            copyright: `Copyright © ${new Date().getFullYear()} Facebook, Inc.`,
          },
        },
      },
    ],
  ]
```
3. Check rendered page, the meta description displays the description value defined in `customFields`:

![image](https://user-images.githubusercontent.com/2687921/75973427-197d9f00-5f10-11ea-9338-26b7a04d0fd5.png)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)
